Framework/Algorithms/src/RemoveExpDecay.cpp

```diff
@@ -153,50 +156,43 @@ void MuonRemoveExpDecay::exec() {
   setProperty("OutputWorkspace", outputWS);
 }
 
-/** This method corrects the errors for one spectra.
+/** This method corrects the errors for one spectrum.
  *	 The muon lifetime is in microseconds not seconds, i.e. 2.1969811 rather
  *   than 0.0000021969811.
  *   This is because the data is in microseconds.
  *   @param inX ::  The X vector
- *   @param inY ::  The input error vector
- *   @param outY :: The output error vector
+ *   @param inE ::  The input error vector
+ *   @param outE :: The output error vector
  */
-void MuonRemoveExpDecay::removeDecayError(const MantidVec &inX,
-                                          const MantidVec &inY,
-                                          MantidVec &outY) {
+void MuonRemoveExpDecay::removeDecayError(const HistogramData::HistogramX &inX,
+                                          const HistogramData::HistogramE &inE,
+                                          HistogramData::HistogramE &outE) {
   // Do the removal
-  for (size_t i = 0; i < inY.size(); ++i) {
-    if (inY[i] != 0.0)
-      outY[i] =
-          inY[i] *
-          exp(inX[i] / (Mantid::PhysicalConstants::MuonLifetime * 1000000.0));
+  for (size_t i = 0; i < inE.size(); ++i) {
+    if (inE[i] != 0.0)
+      outE[i] = inE[i] * exp(inX[i] / MUON_LIFETIME_MICROSECONDS);
     else
-      outY[i] =
-          exp(inX[i] / (Mantid::PhysicalConstants::MuonLifetime * 1000000.0));
+      outE[i] = exp(inX[i] / MUON_LIFETIME_MICROSECONDS);
   }
 }
 
-/** This method corrects the data for one spectra.
+/** This method corrects the data for one spectrum.
  *	 The muon lifetime is in microseconds not seconds, i.e. 2.1969811 rather
  *   than 0.0000021969811.
  *   This is because the data is in microseconds.
  *   @param inX ::  The X vector
  *   @param inY ::  The input data vector
  *   @param outY :: The output data vector
  */
-void MuonRemoveExpDecay::removeDecayData(const MantidVec &inX,
-                                         const MantidVec &inY,
-                                         MantidVec &outY) {
+void MuonRemoveExpDecay::removeDecayData(const HistogramData::HistogramX &inX,
+                                         const HistogramData::HistogramY &inY,
+                                         HistogramData::HistogramY &outY) {
   // Do the removal
   for (size_t i = 0; i < inY.size(); ++i) {
     if (inY[i] != 0.0)
-      outY[i] =
-          inY[i] *
-          exp(inX[i] / (Mantid::PhysicalConstants::MuonLifetime * 1000000.0));
+      outY[i] = inY[i] * exp(inX[i] / MUON_LIFETIME_MICROSECONDS);
     else
-      outY[i] =
-          0.1 *
-          exp(inX[i] / (Mantid::PhysicalConstants::MuonLifetime * 1000000.0));
+      outY[i] = 0.1 * exp(inX[i] / MUON_LIFETIME_MICROSECONDS);
```

Comment:
`removeDecayData` and `removeDecayError` could be done both nicer and cheaper: `exp` is the biggest cost here, so combining both in a single function makes sense. At the same time, the function signature can use `Histogram` instead of the underlying types:

```cpp
// Return by value is cheap for Histogram (and there may be RVO any way)
Histogram MuonRemoveExpDecay::removeDecay(const Histogram &histogram) {
  Histogram result(histogram); // this is cheap and we do not have to worry about copying X
  // remove exp from Y and E here
}
```

The call site will also look cleaner, something like

```cpp
outputWS->setHistogram(i, removeDecay(inputWS->histogram(i));
```

instead of messing around with reading X, Y, and E.